tests/unit/plugins/kubernetes/kubernetes_task_config_test.py

```diff
@@ -228,3 +228,36 @@ def test_valid_node_affinities_invalid_affinity(node_affinity, exc_msg):
             node_affinities=[node_affinity]
         )
     assert exc_msg in exc.value.invariant_errors[0]
+
+
+@pytest.mark.parametrize(
+    "service_account_name", (
+        "",
+        "F" * 300,
```

Comment:
Should we also test one that doesnt' match the DNS regex?

Reply:
done :)